src/hotspot/share/c1/c1_LIRGenerator.cpp

```diff
@@ -1414,13 +1415,15 @@ void LIRGenerator::do_getObjectSize(Intrinsic* x) {
   __ convert(Bytecodes::_i2l, header_size, header_size_long);
   __ add(length, header_size_long, length);
   if (round_mask != 0) {
-    __ logical_and(length, LIR_OprFact::longConst(~round_mask), length);
+    LIR_Opr round_mask_opr = load_immediate(~(jlong)round_mask, T_LONG);
```

Comment:
Explicitly casting to jlong here seems unnecessary.  We don't do that at line 1425.

Reply:
It's going to work for the current value of `round_mask` either way, but I think in general I would cast to an `unsigned long` and then flip the bits. A signed cast will fill the upper bits with ones if the original value was negative. It's usually not what you want with masking.

Reply:
Yes, there is an ambiguity with the old version.  It passes an `int` to longConst(jlong), so the value is signed extended.  If we wanted to treat it as unsigned, we couldn't.

Since we only need the 64-bit T_LONG immediate for _LP64, and only for this one call, I would also be happy with a new load_immediate_64 or load_immediate_jlong that takes a jlong.  Or this section could be changed to use a 32-bit mask and then do the i2l after, like above.  I'm also willing to accept the current changes as is.  I don't have a strong opinion either way.